File: Android/SolNexMobile/app/src/main/java/com/example/solnex/auth/LoginRequest.kt
package com.example.solnex.auth

data class LoginRequest(
    val identifier: String,
    val password: String
)

data class LoginResult(
    val success: Boolean,
    val message: String,
    val token: String? = null,
    val fullName: String? = null,
    val role: String? = null
)

File: backend/SolNex.Api/DTOs/LoginRequest.cs
using System.ComponentModel.DataAnnotations;

namespace SolNex.Api.DTOs;

public sealed class LoginRequest
{
    [Required]
    public string Identifier { get; set; } = string.Empty;

    [Required]
    public string Password { get; set; } = string.Empty;
}

public sealed record LoginResponse(
    string Token,
    DateTime ExpiresAt,
    string Nic,
    string FullName,
    string Email,
    string Role);